admiral/pkg/controller/admiral/rollouts.go

```diff
@@ -194,20 +194,43 @@ func (roc *RolloutController) Deleted(ojb interface{}) {
 	roc.RolloutHandler.Deleted(rollout)
 }
 
-func (d *RolloutController) GetRolloutByLabel(labelValue string, namespace string) []argo.Rollout {
-	matchLabel := common.GetGlobalTrafficDeploymentLabel()
+func (d *RolloutController) GetRolloutBySelectorInNamespace(serviceSelector map[string]string, namespace string) []argo.Rollout {
 	labelOptions := meta_v1.ListOptions{}
-	labelOptions.LabelSelector = fmt.Sprintf("%s=%s", matchLabel, labelValue)
 	matchedRollouts, err := d.RolloutClient.Rollouts(namespace).List(labelOptions)
 
 	if err != nil {
 		logrus.Errorf("Failed to list rollouts in cluster, error: %v", err)
 		return nil
 	}
 
+	var filteredRollouts = make([]argo.Rollout,0)
+
 	if matchedRollouts.Items == nil {
-		return []argo.Rollout{}
+		return filteredRollouts
+	}
+
+	for _, rollout := range matchedRollouts.Items {
+
+		if rollout.Spec.Selector == nil || rollout.Spec.Selector.MatchLabels == nil {
+			continue
+		}
+		match := true
+		for lkey, lvalue := range serviceSelector {
+			// Rollouts controller adds a dynamic label with name rollouts-pod-template-hash to both active and passive replicasets.
+			// This dynamic label is not available on the rollout template. Hence ignoring the label with name rollouts-pod-template-hash
+			if lkey == common.RolloutPodHashLabel {
+				continue
+			}
+			value, ok := rollout.Spec.Selector.MatchLabels[lkey]
```

Comment:
We can use the labelSelector parameter in the list option to find deployments matching given labels.